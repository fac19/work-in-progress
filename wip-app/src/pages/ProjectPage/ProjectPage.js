import React from "react";
import { getProjectPage, getSteps } from "../../utils/get-fetch";
import StepCard from "../../components/StepCard/StepCard";
import { PageHeading } from "../page.style";

const ProjectPage = () => {
  const [projectData, setProjectData] = React.useState([]);
  const [stepsObject, setStepsObject] = React.useState([]);
  // const [feedbackObject, setFeedbackObject] = React.useState([]);

  const projectId = window.location.pathname.replace("/project/", "");

  React.useEffect(() => {
    getProjectPage(projectId).then(setProjectData);
  }, [projectId]);

  React.useEffect(() => {
    getSteps(projectId).then((steps) => {
      setStepsObject(steps);
    });
  }, [projectId]);

  const {
    username,
    project_name,
    project_description,
    project_status,
  } = projectData;

  const makeStepCards = (stepsObject) => {
    return stepsObject
      .map((step) => {
        return <StepCard key={step.id} {...step} />;
      })
      .reverse();
  };

  return (
    <>
      <section>
        <PageHeading>{project_name}</PageHeading>
        <h3>By {username}</h3>
        <p>Project status: {project_status ? "Finished" : "In Progress"}</p>
        <p>{project_description}</p>
      </section>
      <section>
        <h2>Project Steps</h2>
        {/* <a href="">Add a new step</a> */}
        {makeStepCards(stepsObject)}
      </section>
    </>
  );
};

export default ProjectPage;
