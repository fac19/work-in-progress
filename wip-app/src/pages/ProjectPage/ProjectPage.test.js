import React from "react";
import TestRenderer from "react-test-renderer";
import ProjectPage from "./ProjectPage";

describe("Snapshot test", () => {
  it("renders the correct ui", () => {
    const component = TestRenderer.create(<ProjectPage />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
