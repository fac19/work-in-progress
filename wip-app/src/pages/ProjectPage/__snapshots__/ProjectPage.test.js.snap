// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
Array [
  <section>
    <h2 />
    <h3>
      By 
    </h3>
    <p>
      Project status: 
      In Progress
    </p>
    <p />
  </section>,
  <section>
    <h2>
      Project Steps
    </h2>
    <a
      href=""
    >
      Add a new step
    </a>
  </section>,
]
`;
